libs/telemetry/include/telemetry/registry.hpp

```diff
@@ -81,14 +86,33 @@ class Registry
 
 private:
   using MeasurementPtr = std::shared_ptr<Measurement>;
-  using Measurements   = std::vector<MeasurementPtr>;
+  using Measurements =
+      std::unordered_map<std::string, std::unordered_map<std::type_index, MeasurementPtr>>;
```

Comment:
As discussed previously, we need to actually match on name + labels. I think the better solution for this here would be to make an unordered_set of MeasurementPtr with a custom hasher that combines measurement name and labels.

The case that you below allowing more than one type per name is not quite right